drivers/mipi_dsi/dsi_stm32.c

```diff
@@ -0,0 +1,507 @@
+		return -ENOTSUP;
+	}
+
+	if (IS_ENABLED(CONFIG_MIPI_DSI_LOG_LEVEL_DBG)) {
+		char tmp[64];
+
+		if (msg->type == MIPI_DSI_DCS_READ) {
+			snprintk(tmp, sizeof(tmp), "RX: ch %3d, reg 0x%02x, len %2d",
+				 channel, msg->cmd, msg->rx_len);
+			LOG_HEXDUMP_DBG(msg->rx_buf, msg->rx_len, tmp);
+		} else {
+			snprintk(tmp, sizeof(tmp), "TX: ch %3d, reg 0x%02x, len %2d",
+				 channel, msg->cmd, msg->tx_len);
+			LOG_HEXDUMP_DBG(msg->tx_buf, msg->tx_len, tmp);
+		}
+	}
+
+	if (ret != HAL_OK) {
+		LOG_ERR("Transfer failed! (%d)", ret);
+		return -EIO;
+	}
+
+	return len;
+}
+
+static struct mipi_dsi_driver_api dsi_stm32_api = {
+	.attach = mipi_dsi_stm32_attach,
+	.transfer = mipi_dsi_stm32_transfer,
+};
+
+static int mipi_dsi_stm32_init(const struct device *dev)
+{
+	const struct mipi_dsi_stm32_config *config = dev->config;
+	int ret;
+
+	if (!device_is_ready(config->rcc)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(config->rcc, (clock_control_subsys_t)&config->dsi_clk);
+	if (ret < 0) {
+		LOG_ERR("Enable DSI peripheral clock failed! (%d)", ret);
+		return ret;
+	}
+
+	reset_line_toggle_dt(&config->reset);
+
+	ret = mipi_dsi_stm32_host_init(dev);
+	if (ret) {
+		LOG_ERR("Setup DSI host failed! (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+#define CHILD_GET_DATA_LANES(child) DT_PROP(child, data_lanes)
+
+#define STM32_MIPI_DSI_DEVICE(inst)								\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, host_timeouts),					\
+		(static DSI_HOST_TimeoutTypeDef host_timeouts_##inst = {			\
+			.TimeoutCkdiv = DT_INST_PROP_BY_IDX(inst, host_timeouts, 0),		\
+			.HighSpeedTransmissionTimeout =						\
+				DT_INST_PROP_BY_IDX(inst, host_timeouts, 1),			\
+			.LowPowerReceptionTimeout =						\
+				DT_INST_PROP_BY_IDX(inst, host_timeouts, 2),			\
+			.HighSpeedReadTimeout = DT_INST_PROP_BY_IDX(inst, host_timeouts, 3),	\
+			.LowPowerReadTimeout = DT_INST_PROP_BY_IDX(inst, host_timeouts, 4),	\
+			.HighSpeedWriteTimeout = DT_INST_PROP_BY_IDX(inst, host_timeouts, 5),	\
+			.HighSpeedWritePrespMode = DT_INST_PROP_BY_IDX(inst, host_timeouts, 6),	\
+			.LowPowerWriteTimeout = DT_INST_PROP_BY_IDX(inst, host_timeouts, 7),	\
+			.BTATimeout = DT_INST_PROP_BY_IDX(inst, host_timeouts, 8)		\
+		}), ());									\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, phy_timings),					\
+		(static DSI_PHY_TimerTypeDef phy_timings_##inst = {				\
+			.ClockLaneHS2LPTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 0),	\
+			.ClockLaneLP2HSTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 1),	\
+			.DataLaneHS2LPTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 2),		\
+			.DataLaneLP2HSTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 3),		\
+			.DataLaneMaxReadTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 4),	\
+			.StopWaitTime = DT_INST_PROP_BY_IDX(inst, phy_timings, 5)		\
+		}), ());									\
+	static const uint32_t data_lanes_##inst[] =						\
+		DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(inst, CHILD_GET_DATA_LANES, (,));		\
```

Comment:
`DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP_VARGS(inst, DT_PROP, (,), data_lanes)` should work, so you don't need an extra define

Reply:
I think I should then use `DT_PROP_BY_IDX` as in the display drivers.
Unfortunately it depends then on the usage of the `data-lanes` property in the display driver,  which is probably not always at index `0`. Example from `rm68200.c`: `DT_INST_PROP_BY_IDX(inst, data_lanes, 0)`

What is the intention of defining `data-lanes` as an array?
Is it intended to be used as lane swap, see https://github.com/torvalds/linux/blob/7e364e56293bb98cae1b55fd835f5991c4e96e7d/Documentation/devicetree/bindings/media/video-interfaces.yaml#L159 ?